crypto/x509v3/v3_ncons.c

```diff
@@ -297,47 +297,143 @@ int NAME_CONSTRAINTS_check(X509 *x, NAME_CONSTRAINTS *nc)
 
 }
 
+#define CN2DNSID_NOMEM   (-2)
+#define CN2DNSID_BADNAME (-1)
+
+static int cn2dnsid(ASN1_STRING *cn, unsigned char **dnsid)
+{
+    int utf8_length;
+    int i;
+    unsigned char *utf8_value;
+    int isdnsname = 0;
+
+    if ((utf8_length = ASN1_STRING_to_UTF8(&utf8_value, cn)) < 0)
```

Comment:
How does this work with internationalised domain names?

Reply:
Why do we do this conversion to UTF8 here?

Reply:
Adding comments to answer these questions, the logic is correct and will remain the same.